silence/silence_test.go

```diff
@@ -465,8 +465,12 @@ func TestSilenceLimits(t *testing.T) {
 			MaxSilences:        1,
 			MaxPerSilenceBytes: 2 << 11, // 4KB
 		},
+		Retention: 100 * time.Millisecond,
 	})
 	require.NoError(t, err)
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go s.Maintenance(100*time.Millisecond, "", stopCh, nil)
```

Comment:
We cannot force a maintenance without running the goroutine, hence the calls to `time.Sleep`. I think this is fine, and I can improve the tests later.